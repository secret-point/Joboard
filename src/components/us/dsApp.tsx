import React from "react";
import { connect } from "react-redux";
import { BackToTopButton } from '@amzn/stencil-react-components/back-to-top-button';
import { HashRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import PreConsentPage from "./preConsent/PreConsent";
import ConsentPage from "./consent/Consent";
import { AppConfig } from "../../@types/IPayload";
import { PAGE_ROUTES } from "../pageRoutes";
import JobOpportunity from "./jobOpportunity/JobOpportunity";
import { Col } from "@amzn/stencil-react-components/layout";
import AppLoader from "../common/AppLoader";
import JobConfirmation from "./jobOpportunity/JobConfirmation";
import JobDescription from "./jobOpportunity/JobDescription";
import ContingencyOffer from "./contingentOffer/ContingentOffer";
import BackgroundCheck from "./bgc/BackgroundCheck";
import FcraDisclosure from "../common/bgc/FcraDisclosure";
import ResumeApplication from "./resumeApplication/ResumeApplication";
import Nhe from "./nhe/Nhe";
import SelfIdentification from "./selfIdentification/SelfIdentification";
import AlreadyApplied from "./alreadyApplied/AlreadyApplied";
import ReviewSubmit from "./reviewSubmit/ReviewSubmit";
import ThankYou from "./thankYou/ThankYou";
import CounterMessageBanner from "../common/CounterMessageBanner";
import SessionTimeout from "./sessionTimeout/SessionTimeout";
import WOTCComponent from "./wotc/WOTC";
import AssessmentConsent from "./assessment/AssessmentConsent";
import AssessmentNotEligible from "./assessment/AssementNotEligible";
import AssessmentFinished from "./assessment/AssessmentFinished";
import CandidateWithdraws from "../common/CandidateWithdraws";
import { BannerMessage } from "../common/BannerMessage";
import { uiState } from "../../reducers/ui.reducer";
import CaliDisclosure from "./caliDisclosure/CaliDisclosure";
import AmazonRejects from "./amazonRejects/AmazonRejects";

interface MapStateToProps {
  appConfig: AppConfig,
  ui: uiState
}

const {
  ALREADY_APPLIED,
  ASSESSMENT_CONSENT,
  ASSESSMENT_FINISHED,
  ASSESSMENT_NOT_ELIGIBLE,
  BACKGROUND_CHECK,
  BACKGROUND_CHECK_FCRA,
  CANDIDATE_WITHDRAWS,
  CONSENT,
  CONTINGENT_OFFER,
  JOB_CONFIRMATION,
  JOB_DESCRIPTION,
  JOB_OPPORTUNITIES,
  NHE,
  PRE_CONSENT,
  RESUME_APPLICATION,
  REVIEW_SUBMIT,
  SELF_IDENTIFICATION,
  SESSION_TIMEOUT,
  THANK_YOU,
  WOTC,
  CALI_DISCLOSURE,
  AMAZON_REJECTS
} = PAGE_ROUTES;

const DragonStoneAppUS = ( props: MapStateToProps ) => {
  const { appConfig, ui } = props;
  const { bannerMessage } = ui;

  // Will review and refacoter this logic later
  // const onClick = () => {
  //   const isCandidateDashboardEnabled = appConfig.featureList?.CANDIDATE_DASHBOARD?.isAvailable;
  //   const queryParamsInSession = window.sessionStorage.getItem("query-params");
  //   const queryParams = queryParamsInSession
  //     ? JSON.parse(queryParamsInSession)
  //     : {};
  //   const queryStringFor3rdParty = get3rdPartyFromQueryParams(queryParams,'?');
  //   const candidateDashboardUrl = `${appConfig.CSDomain}/app${queryStringFor3rdParty}#/myApplications`;
  //   window.location.assign(isCandidateDashboardEnabled? candidateDashboardUrl : appConfig.dashboardUrl);
  // };

  return (
      <Col padding='S300'>
        <AppLoader/>
        <CounterMessageBanner/>
        {!!bannerMessage && <BannerMessage bannerMessage={bannerMessage}/>}
        <Router>
          <Switch>
            <Route exact path='/' render={() => <Redirect to={PRE_CONSENT}/>}/>
            <Route path={`/${PRE_CONSENT}`} exact>
              <PreConsentPage/>
            </Route>
            <Route path={`/${CONSENT}`} exact>
              <ConsentPage/>
            </Route>
            <Route path={`/${JOB_OPPORTUNITIES}`} exact>
              <JobOpportunity/>
            </Route>
            <Route path={`/${JOB_CONFIRMATION}`} exact>
              <JobConfirmation/>
            </Route>
            <Route path={`/${JOB_DESCRIPTION}`} exact>
              <JobDescription/>
            </Route>
            <Route path={`/${CONTINGENT_OFFER}`} exact>
              <ContingencyOffer/>
            </Route>
            <Route path={`/${BACKGROUND_CHECK}`} exact>
              <BackgroundCheck/>
            </Route>
            <Route path={`/${BACKGROUND_CHECK_FCRA}`} exact>
              <FcraDisclosure/>
            </Route>
            <Route path={`/${RESUME_APPLICATION}`} exact>
              <ResumeApplication/>
            </Route>
            <Route path={`/${NHE}`} exact>
              <Nhe/>
            </Route>
            <Route path={`/${SELF_IDENTIFICATION}`} exact>
              <SelfIdentification/>
            </Route>
            <Route path={`/${ALREADY_APPLIED}`} exact>
              <AlreadyApplied/>
            </Route>
            <Route path={`/${REVIEW_SUBMIT}`} exact>
              <ReviewSubmit/>
            </Route>
            <Route path={`/${THANK_YOU}`} exact>
              <ThankYou/>
            </Route>
            <Route path={`/${WOTC}`} exact>
              <WOTCComponent/>
            </Route>
            <Route path={`/${SESSION_TIMEOUT}`} exact>
              <SessionTimeout/>
            </Route>
            <Route path={`/${ASSESSMENT_CONSENT}`} exact>
              <AssessmentConsent/>
            </Route>
            <Route path={`/${ASSESSMENT_NOT_ELIGIBLE}`} exact>
              <AssessmentNotEligible/>
            </Route>
            <Route path={`/${ASSESSMENT_FINISHED}`} exact>
              <AssessmentFinished/>
            </Route>
            <Route path={`/${CANDIDATE_WITHDRAWS}`} exact>
              <CandidateWithdraws/>
            </Route>
            <Route path={`/${CALI_DISCLOSURE}`} exact>
              <CaliDisclosure/>
            </Route>
            <Route path={`/${AMAZON_REJECTS}`} exact>
              <AmazonRejects/>
            </Route>
          </Switch>
        </Router>
        <BackToTopButton />
      </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(DragonStoneAppUS);
