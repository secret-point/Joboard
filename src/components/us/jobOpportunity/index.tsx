import React, { useEffect } from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import JobOpportunity from "./JobOpportunity";
import JobConfirmation from "./JobConfirmation";
import { IconArrowLeft, IconSize } from "@amzn/stencil-react-components/icons";
import { Text } from "@amzn/stencil-react-components/text";
import { JOB_OPPORTUNITY_PAGE } from "../../../utils/enums/common";
import { CommonColors } from "../../../utils/colors";
import JobDescription from "./JobDescription";
import { translate as t } from "../../../utils/translator";
import { connect } from "react-redux";
import { uiState } from "../../../reducers/ui.reducer";
import { boundSetJobOpportunityPage } from "../../../actions/UiActions/boundUi";

interface MapStateToProps {
    ui: uiState
}

const JobOpportunityContainer = (props: MapStateToProps) => {
    const { ui } = props;
    const { jobOpportunity } = ui;
    const { currentPage } = jobOpportunity;
    const { INDEX, JOB_CONFIRMATION, JOB_DESCRIPTION } = JOB_OPPORTUNITY_PAGE;

    useEffect(() => {
        window.scroll(0, 0);
    })

    const renderPage = ( pageType: JOB_OPPORTUNITY_PAGE ) => {
        switch(pageType) {
            case INDEX:
                return <JobOpportunity/>;

            case JOB_CONFIRMATION:
                return <JobConfirmation />;

            case JOB_DESCRIPTION:
                return <JobDescription/>;

            default:
                return <JobOpportunity />
        }
    }


    const handleBack = () => {
        if(currentPage === JOB_CONFIRMATION) {
            boundSetJobOpportunityPage(INDEX);
        }
        else {
            boundSetJobOpportunityPage(JOB_CONFIRMATION);
        }
    }

    return (
        <Col>
            {
                currentPage !== INDEX &&
                <Row
                    gridGap={5}
                    alignItems="center"
                    width="fit-content"
                    color={CommonColors.Blue70}
                    padding='S200'
                    onClick={handleBack}
                    style={{ cursor: 'pointer' }}
                >
                    <IconArrowLeft size={IconSize.ExtraSmall}/>
                    <Text fontSize="T100" fontWeight="medium">
                        {currentPage === JOB_CONFIRMATION ?
                            t('BB-JobOpportunity-back-to-indexPage-link', 'View Jobs') :
                            t('BB-JobOpportunity-back-to-jobConfirmation-link','Back')
                        }
                    </Text>
                </Row>
            }
            {
                renderPage(currentPage)
            }
        </Col>
    )
}

const mapStateToProps = ( state: MapStateToProps ) => {
    return state;
};

export default connect(mapStateToProps)(JobOpportunityContainer)
