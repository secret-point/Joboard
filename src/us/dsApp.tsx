import React from "react";
import { connect } from "react-redux";
import {
  HashRouter as Router,
  useLocation,
  Switch,
  Route
} from "react-router-dom";
import PreConsentPage from "./preConsent/PreConsent";
import ConsentPage from "./consent/Consent";
import queryString from "query-string";
import CandidateApplicationService from "../services/candidate-application-service";
import { CreateApplicationRequestDS } from "../@types/candidate-application-service-requests";
import moment from "moment";
import { StencilProvider } from "@amzn/stencil-react-components/dist/submodules/context";
import { checkIfIsCSRequest, get3rdPartyFromQueryParams } from "../helpers/utils";
import { AppConfig } from "../@types/IPayload";
import store from "../store/store";

interface MapStateToProps {
  appConfig: AppConfig,
  app: any
}

const DragonStoneAppUS = (props: MapStateToProps) => {
  const { appConfig } = props;
  console.log("appConfig", appConfig, store.getState());

  const onClick = () => {
    const isCandidateDashboardEnabled = appConfig.featureList?.CANDIDATE_DASHBOARD?.isAvailable;
    const queryParamsInSession = window.sessionStorage.getItem("query-params");
    const queryParams = queryParamsInSession
      ? JSON.parse(queryParamsInSession)
      : {};
    const queryStringFor3rdParty = get3rdPartyFromQueryParams(queryParams,'?');
    const candidateDashboardUrl = `${appConfig.CSDomain}/app${queryStringFor3rdParty}#/myApplications`;
    window.location.assign(isCandidateDashboardEnabled? candidateDashboardUrl : appConfig.dashboardUrl);
  };

  return (
    <StencilProvider>
      <Router>
        <Switch>
          <Route path="/consent/">
            <ConsentPage />
          </Route>
          <Route path="/">
            <PreConsentPage />
          </Route>
        </Switch>
      </Router>
    </StencilProvider>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(DragonStoneAppUS);
