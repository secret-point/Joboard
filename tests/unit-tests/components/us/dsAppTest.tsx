import React from "react";
import { shallow } from "enzyme";
import { DragonStoneAppUS } from "../../../../src/components/us/dsApp";
import { TEST_APP_CONFIG, TestInitUiState } from "../../../test-utils/test-data";

describe("App", () => {
  test("snapshot", () => {
    const wrapper = shallow(<DragonStoneAppUS appConfig={TEST_APP_CONFIG} ui={TestInitUiState} />);
    expect(wrapper).toMatchSnapshot();
  });
});
