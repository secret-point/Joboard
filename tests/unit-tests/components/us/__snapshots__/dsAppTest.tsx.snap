// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<Col
  minHeight="40vh"
  padding="S300"
>
  <Connect(AppLoader) />
  <Connect(CounterMessageBanner) />
  <HashRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/pre-consent"
      >
        <Connect(PreConsent) />
      </Route>
      <Route
        exact={true}
        path="/consent"
      >
        <Connect(Consent) />
      </Route>
      <Route
        exact={true}
        path="/job-opportunities"
      >
        <Connect(JobOpportunity) />
      </Route>
      <Route
        exact={true}
        path="/job-opportunities/job-confirmation"
      >
        <Connect(JobConfirmation) />
      </Route>
      <Route
        exact={true}
        path="/job-opportunities/job-description"
      >
        <Connect(JobDescription) />
      </Route>
      <Route
        exact={true}
        path="/contingent-offer"
      >
        <Connect(ContingentOffer) />
      </Route>
      <Route
        exact={true}
        path="/bgc"
      >
        <Connect(BackgroundCheck) />
      </Route>
      <Route
        exact={true}
        path="/bgc/fcra"
      >
        <Connect(FcraDisclosure) />
      </Route>
      <Route
        exact={true}
        path="/resume-application"
      >
        <Connect(ResumeApplication) />
      </Route>
      <Route
        exact={true}
        path="/nhe"
      >
        <Connect(Nhe) />
      </Route>
      <Route
        exact={true}
        path="/self-identification"
      >
        <Connect(SelfIdentification) />
      </Route>
      <Route
        exact={true}
        path="/already-applied"
      >
        <Connect(AlreadyApplied) />
      </Route>
      <Route
        exact={true}
        path="/review-submit"
      >
        <Connect(ReviewSubmit) />
      </Route>
      <Route
        exact={true}
        path="/thank-you"
      >
        <Connect(ThankYou) />
      </Route>
      <Route
        exact={true}
        path="/wotc"
      >
        <Connect(Wotc) />
      </Route>
      <Route
        exact={true}
        path="/wotc-complete"
      >
        <Connect(WotcComplete) />
      </Route>
      <Route
        exact={true}
        path="/session-timeout"
      >
        <Connect(SessionTimeout) />
      </Route>
      <Route
        exact={true}
        path="/assessment-consent"
      >
        <Connect(AssessmentConsent) />
      </Route>
      <Route
        exact={true}
        path="/assessment-not-eligible"
      >
        <Connect(AssessmentNotEligible) />
      </Route>
      <Route
        exact={true}
        path="/assessment-finished"
      >
        <Connect(AssessmentFinished) />
      </Route>
      <Route
        exact={true}
        path="/candidate-withdraws"
      >
        <Connect(CandidateWithdraws) />
      </Route>
      <Route
        exact={true}
        path="/cali-disclosure"
      >
        <Connect(CaliDisclosure) />
      </Route>
      <Route
        exact={true}
        path="/amazon-rejects"
      >
        <Connect(AmazonRejects) />
      </Route>
      <Route
        exact={true}
        path="/workflow-failed"
      >
        <Connect(WorkflowFailed) />
      </Route>
      <Route
        exact={true}
        path="/amazon-withdraws"
      >
        <Connect(AmazonWithdraws) />
      </Route>
      <Route
        exact={true}
        path="/no-available-time-slots"
      >
        <Connect(NoAvailableTimeSlots) />
      </Route>
      <Route
        exact={true}
        path="/can-not-offer-job"
      >
        <Connect(CanNotOfferJob) />
      </Route>
      <Route
        exact={true}
        path="/supplementary-success"
      >
        <Connect(SupplementarySuccess) />
      </Route>
      <Route
        exact={true}
        path="/rehire-eligibility-status"
      >
        <Connect(RehireEligibilityStatus) />
      </Route>
      <Route
        exact={true}
        path="/no-available-shift"
      >
        <Connect(NoAvailableShift) />
      </Route>
      <Route
        exact={true}
        path="/applicationId-null"
      >
        <Connect(ApplicationIdNull) />
      </Route>
    </Switch>
  </HashRouter>
  <BackToTopButton
    altText="Back to top"
    bottom="32px"
    scrollOffset={250}
    trailing="24px"
  />
</Col>
`;
